### PR TITLE
Coadd obs

### PR DESCRIPTION
**Pull request by ngbusca**
This PR refactors the way spCFrames are read:

* each spCFrame file is open once (so there might be a slight gain in performace)
* only spCFrames participating to an spPlate file are used (I don't know whether all spCFrames can be rejected, but if they are, then they won't be used)
but mainly:
* the user can choose to use the spCFrames participating to the best obs or all (reproducing what ForetFusion does but directly at the spCFrame level and without the intermediate wavelength resampling)

### CODE REVIEW

File: bin/do_deltas.py [hunk 1]
```diff
@@ -170,10 +176,9 @@ def cont_fit(data):
     nit = args.nit
 
     log = open(args.log,'w')
-    data,ndata = io.read_data(args.in_dir,args.drq,args.mode,\
-                              zmin=args.zqso_min,zmax=args.zqso_max,nspec=args.nspec,log=log,keep_bal=args.keep_bal,bi_max = args.bi_max,order=args.order)
-    
-
+    data, ndata = io.read_data(args.in_dir,args.drq,args.mode,\
+                              zmin=args.zqso_min,zmax=args.zqso_max,nspec=args.nspec,log=log,keep_bal=args.keep_bal,bi_max = args.bi_max,order=args.order, best_obs = args.best_obs, single_exp = args.single_exp)
```
Comment (londumas) [inline bin/do_deltas.py]: Could you split this line on more lines?

File: py/picca/io.py [hunk 2]
```diff
@@ -178,19 +200,12 @@ def read_data(in_dir,drq,mode,zmin = 2.1,zmax = 3.5,nspec=None,log=None,keep_bal
...
-            pix_data = read_from_spcframe(in_dir,thid[w], ra[w], dec[w], zqso[w], plate[w], mjd[w], fid[w], order, mode=mode, log=log)
-            read_time=time.time()-t0
         if not pix_data is None:
             sys.stderr.write("{} read from pix {}, {} {} in {} secs per spectrum\n".format(len(pix_data),pix,i,len(upix),read_time/(len(pix_data)+1e-3)))
         if not pix_data is None and len(pix_data)>0:
             data[pix] = pix_data
             ndata += len(pix_data)
 
-        if not nspec is None:
-            if ndata > nspec:break
```
Comment (londumas) [inline py/picca/io.py]: why do you remove this?

File: py/picca/io.py [hunk 3]
```diff
@@ -262,51 +277,111 @@ def read_from_pix(in_dir,pix,thid,ra,dec,zqso,plate,mjd,fid,order,log=None):
...
 
-                index = fib_list.index(f)
+            flux = spcframe[0].read()
+            ivar = spcframe[1].read()*(spcframe[2].read()==0)
+            llam = spcframe[3].read()
+            
+            ## now convert all those fluxes into forest objects
+            for index, (t, r, d, z, p, m, f) in enumerate(zip(thid[wfib], ra[wfib], dec[wfib], zqso[wfib], plate[wfib], mjd[wfib], fid[wfib])):
+                index =(f-1)%500
+                wlam = (10**llam[index]/(1+z) > 1220) & (10**llam[index]/(1+z)<1230)
```
Comment (londumas) [inline py/picca/io.py]: why 1220 and 1230? and I don't see where you use this.

File: py/picca/io.py (hunk 2 above)
Comment (ngbusca) [reply py/picca/io.py]: this is now done between l. 116 and l.124:

    if nspec != None:
        ra = ra[:nspec]
        dec = dec[:nspec]
        zqso = zqso[:nspec]
        thid = thid[:nspec]
        plate = plate[:nspec]
        mjd = mjd[:nspec]
        fid = fid[:nspec]

File: bin/do_deltas.py (hunk 1 above)
Comment (ngbusca) [reply bin/do_deltas.py]: sure

File: py/picca/io.py (hunk 3 above)
Comment (ngbusca) [reply py/picca/io.py]: woops, leftover from debugging...

File: bin/do_deltas.py (hunk 1 above)
Comment (ngbusca) [reply bin/do_deltas.py]: done

File: py/picca/io.py (hunk 3 above)
Comment (ngbusca) [reply py/picca/io.py]: removed

**Pull request merged**